### PR TITLE
Support for application-level caching via an lru cache with ttl

### PR DESCRIPTION
**Pull request by wojcikstefan**

### CODE REVIEW

File: flask_ipblock/__init__.py [hunk 1]
```diff
@@ -42,5 +55,27 @@ def block_before(self):
...
             return 'IP Blocked', 200
+
+    def matches_ip(self, ip):
+        """Return True if the given IP is blacklisted, False otherwise."""
+
+        # Check the cache if caching is enabled
+        if self.cache is not None:
+            matches_ip = self.cache.get(ip)
+            if matches_ip is not None:
+                if matches_ip:
```
Comment (thomasst) [inline flask_ipblock/__init__.py]: `return matches_ip`

File: flask_ipblock/__init__.py [hunk 2]
```diff
@@ -3,15 +3,28 @@
...
         You can override the default MongoDB read preference via the
         optional read_preference kwarg.
+
+        You can limit the impact of the IP checks on your MongoDB by
+        maintaining a local in-memory LRU cache. To do so, specify its
+        size via cache_size and the TTL (in seconds) via cache_ttl.
         """
         self.read_preference = read_preference
+
+        self.cache = None
```
Comment (thomasst) [inline flask_ipblock/__init__.py]: Can just put that in an `else` clause below (no need to double assign to `self.cache` if there's a cache)

**Review by thomasst — CHANGES_REQUESTED** [review]

File: flask_ipblock/__init__.py [hunk 3]
```diff
@@ -3,15 +3,28 @@
...
         """
         Initialize IPBlock and set up a before_request handler in the
         app.
 
         You can override the default MongoDB read preference via the
         optional read_preference kwarg.
+
+        You can limit the impact of the IP checks on your MongoDB by
+        maintaining a local in-memory LRU cache. To do so, specify its
+        size via cache_size and the TTL (in seconds) via cache_ttl.
```
Comment (thomasst) [inline flask_ipblock/__init__.py]: Clarify that size = number of IP addresses, not storage in bytes.

**Comment by wojcikstefan** [comment]
@thomasst should be good now, could you take another look?

**Review by thomasst — APPROVED** [review]

**Pull request merged**